apps/backend/src/schemas/configuration.py

```
@@ -0,0 +1,549 @@
+        if len(v.strip()) == 0:
+            raise ValueError("Raw content cannot be empty")
+        if len(v) > 10485760:  # 10MB max
+            raise ValueError("Raw content cannot exceed 10MB")
+        return v
+
+    @model_validator(mode="after")
+    def validate_configuration_consistency(self):
+        """Cross-field validation for configuration snapshot consistency"""
+        # Validate change type logic
+        if self.change_type == "deleted" and self.raw_content:
+            if len(self.raw_content.strip()) > 0:
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)